core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -67,23 +67,23 @@ class TopicConfigHandler(private val logManager: LogManager, kafkaConfig: KafkaC
       logs.foreach(_.config = logConfig)
     }
 
-    val brokerId = kafkaConfig.brokerId
-
-    if (topicConfig.containsKey(LogConfig.ThrottledReplicasListProp) && topicConfig.getProperty(LogConfig.ThrottledReplicasListProp).length > 0) {
-      val partitions = parseThrottledPartitions(topicConfig, brokerId)
-      quotas.leader.markThrottled(topic, partitions)
-      quotas.follower.markThrottled(topic, partitions)
-      logger.debug(s"Setting throttled partitions on broker $brokerId for topic: $topic and partitions $partitions")
-    } else {
-      quotas.leader.removeThrottle(topic)
-      quotas.follower.removeThrottle(topic)
-      logger.debug(s"Removing throttled partitions from broker $brokerId for topic $topic")
+    def updateThrottledList(prop: String, quotaManager: ReplicationQuotaManager) = {
+      if (topicConfig.containsKey(prop) && topicConfig.getProperty(prop).length > 0) {
+        val partitions = parseThrottledPartitions(topicConfig, kafkaConfig.brokerId, prop)
+        quotaManager.markThrottled(topic, partitions)
+        logger.debug(s"Setting $prop on broker ${kafkaConfig.brokerId} for topic: $topic and partitions $partitions")
```

Comment:
This is built into the property, which is why I didn't add it explicitly. That should suffice no?